git/05.merge-strategies.md

```diff
@@ -0,0 +1,116 @@
+## [Home](../README.md)
+
+## 5. Merge Strategies
+
+Once your feature branch is approved by a reviewer and ready to merge, how do
+you merge it into the [release branch](07.release-branching.md)?
+
+The [Git Development Workflow](02.development-workflow.md) best practice
+describes the process at a high level, including rebasing, squashing, merging,
+pushing, and cleaning up. This document provides more detail about the merge
+step.
+
+### Background
+
+There are two types of merges: fast-forward (FF) merges and non-fast-forward
+(non-FF) merges. Also, non-FF merges can be _linearizable_, meaning there is no
+ambiguity about the logically linear ordering of commits. Note that linearizable
+non-FF merges can be reduced to a FF merge, a property that we discuss below.
+
+Let's consider the difference with an example. Say this is your starting point
+(ignoring the complexity of a release branch for now):
+
+```
+A   B   C
+o---o---o master
+     \
+      F   G
+      o---o feature
+```
+
+A non-linearizable non-FF merge of `feature` into `master` would look like this:
+
+```
+A   B   C    M
+o---o---o---o master
+     \     /
+      F   G
+      o---o feature
+```
+
+Here `M` is a _merge commit_, which is a commit with more than one parent
+commit. Note that even if the `feature` branch is deleted, the structure of the
+commit history remains the same, and `M` is still needed.
+
+The other two examples require rebasing `feature` onto master first, resulting
```

Comment:
Hmm, I think personally I'd prefer not to. I think it reads well as is without getting distracted by implementation details, and such details are (1) probably known by most already and (2) easy to lookup. And I think if we added a note here it'd make sense to add similar notes throughout the document.

…but I'm willing, if despite my reasons you'd like me to. :-)